framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1345,7 +1345,11 @@ private AnnotatedTypeMirror fromExpression(ExpressionTree tree) {
 
         AnnotatedTypeMirror result = TypeFromTree.fromExpression(this, tree);
 
-        if (shouldCache) {
+        if (shouldCache && tree.getKind() != Tree.Kind.NEW_CLASS) {
+            // Don't cache types of NEW operator
+            // Cached NEW_CLASS types may contain "Unknowned*" annotations
```

Comment:
The cached types may contain incorrect annotations that were computed during dataflow.  The better solution is to only clear the cache if dataflow is running.

Reply:
@smillst Thanks for the pointer to #602!
@abslamp Can you look into disabling caching during dataflow analysis and fixing #602?

Reply:
I will change the comments for now, but I want to look into the better solution.